kivy/graphics/context_instructions.pyx

```diff
@@ -764,9 +764,25 @@ cdef class Scale(Transform):
     .. deprecated:: 1.6.0
         Deprecated single scale property in favor of x, y, z, xyz axis
         independent scaled factors.
+
+    .. versionchanged:: 2.3.0
+        Allowed kwargs to be used to supply x, y and z.
     '''
     def __init__(self, *args, **kwargs):
         cdef double x, y, z
+
+        x, y, z = 1.0, 1.0, 1.0
+        if len(args) == 1:
+            s = args[0]
+            x, y, z = s, s, s
+        if len(args) == 2:
+            x, y = args
+        elif len(args) == 3:
+            x, y, z = args
```

Comment:
Referring to my previous comment, I think we should only consider the `len(args) == 3` option.
Also, by removing the deprecated single-scale property, we can likely clean up the code a lot. 😀

Reply:
I have done this, and replaced the deprecation notice with a versionchanged message.